endogenous/scripts/end_transport.py

```diff
@@ -164,36 +222,79 @@ def transport(string):
     df_load = pd.read_csv(string + 'resources/transport_demand_s_45.csv', sep=',', index_col=0) 
     df_load.index = pd.to_datetime(df_load.index, utc=True)
     df_load.index.name = 'utc_time'
-    load_p = df_load['AT1 0']
+    load_p = df_load['DK1 0']
     # data for load: resources/transport_demand_s{simpl}_{clusters}.csv 
-    
+
     network.add("Load",
                 "load2",
                 bus = "vehicle bus",
                 carrier = "vehicle",
                 p_set = load_p)
+        
+    
+    network.add(
+        "GlobalConstraint",
+        "co2_limit",
+        sense = "<=",
+        carrier_attribute = "co2_emissions",
+        constant = co2_limit,
+    )
     
     #print(network.loads_t.p_set)
+    print(network.carriers)
     
-    network.lopf(network.snapshots, 
+    status, status2 = network.lopf(network.snapshots, 
                  pyomo=False,
                  solver_name='gurobi')
     
-    # print objective value
-    print("Objective value: %f" % network.objective)
-
-    # print oil generator value
-    print("Oil generator value: %f" % network.generators.p_nom_opt["oil"])
+    if(status == 'ok'):
+        # print objective value
+        print("Objective value: %f" % network.objective)
+    
+        # print oil generator value
+        print("Oil generator value: %f" % network.generators.p_nom_opt["oil"])
+    
+        # print solar generator value
+        print("Solar generator value: %f" % network.generators.p_nom_opt["solar"])
+    
+        network.generators_t.p.plot()
```

Comment:
This output figure is great but try to generate additional figures that help us understand if the endogenous transport is working properly, e.g
[ ] Evolution of the land transport demand supplied with electricity and oil throughout the different planning horizons
[ ] Evolution of the number of EVs and ICE cars throughout the different planning horizons
[ ] Evolution of the filling levels of the EV battery for different planning horizons

Reply:
Added plot for generators and ratio of EV and ICEs in commit bcf96f98b34076525d7292b7f6105ddb630ac759

Reply:
In the output figure 'generators.png'. what are the units on the y axis?

In the output figure 'vehiclenb.png'. what are the units on the y axis? why are the EVs decreasing? shouldn't it be the opposite?

Can you add some plots to verify that the results make sense?

Also, can you add some plots to verify the charging pattern/filling levels of the EVs? E.g.  you can plot, the average (for every day in a year) of the filling level of the EV battery at every hour of the day, or a heatmap such as the one in Figure 7 in https://doi.org/10.1016/j.enconman.2019.111977 (code [here](https://github.com/martavp/PyPSA-plots/blob/master/scripts/4_plot_charge_discharge_hotmaps.py)


In short, try to plot results so that you can show that the code is working

Reply:
I see now that new figures were added by @ebbekyhl, super!